pandas/core/arrays/datetimes.py

```diff
@@ -194,6 +201,117 @@ def __new__(cls, values, freq=None, tz=None):
+            # sub-Day Tick
+            if inferred_tz is None and tz is not None:
+                # naive dates
+                if start is not None and start.tz is None:
+                    start = start.tz_localize(tz, ambiguous=False)
+
+                if end is not None and end.tz is None:
+                    end = end.tz_localize(tz, ambiguous=False)
+
+            if start and end:
+                if start.tz is None and end.tz is not None:
+                    start = start.tz_localize(end.tz, ambiguous=False)
+
+                if end.tz is None and start.tz is not None:
+                    end = end.tz_localize(start.tz, ambiguous=False)
+
+            if cls._use_cached_range(freq, _normalized, start, end):
+                index = cls._cached_range(start, end, periods=periods,
+                                          freq=freq)
+            else:
+                index = _generate_regular_range(cls, start, end, periods, freq)
+
+        else:
+
+            if tz is not None:
+                # naive dates
+                if start is not None and start.tz is not None:
+                    start = start.replace(tzinfo=None)
+
+                if end is not None and end.tz is not None:
+                    end = end.replace(tzinfo=None)
+
+            if start and end:
+                if start.tz is None and end.tz is not None:
+                    end = end.replace(tzinfo=None)
+
+                if end.tz is None and start.tz is not None:
+                    start = start.replace(tzinfo=None)
+
+            if freq is not None:
+                if cls._use_cached_range(freq, _normalized, start, end):
+                    index = cls._cached_range(start, end, periods=periods,
+                                              freq=freq)
+                else:
+                    index = _generate_regular_range(cls, start, end,
+                                                    periods, freq)
+
+                if tz is not None and getattr(index, 'tz', None) is None:
+                    arr = conversion.tz_localize_to_utc(ensure_int64(index.values),
+                                                        tz,
+                                                        ambiguous=ambiguous)
+
+                    index = cls(arr)
+
+                    # index is localized datetime64 array -> have to convert
+                    # start/end as well to compare
+                    if start is not None:
+                        start = start.tz_localize(tz).asm8
+                    if end is not None:
+                        end = end.tz_localize(tz).asm8
+            else:
+                # Create a linearly spaced date_range in local time
+                start = start.tz_localize(tz)
+                end = end.tz_localize(tz)
+                arr = np.linspace(start.value, end.value, periods)
+                index = cls._simple_new(arr.astype('M8[ns]'), freq=None, tz=tz)
```

Comment:
One of two things changed from the existing version.  ATM this is:

```
                index = tools.to_datetime(np.linspace(start.value,
                                                      end.value, periods),
                                          utc=True)
                index = index.tz_convert(tz)
```

Changed to avoid the dependency on `tools`.